### PR TITLE
refactor!: eliminate sdk codec dependency from remote mode

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR eliminate the codec dependency from remote node to make Juno chain-agnostic.

## Checklist
- [x] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.  
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: cmd/parse/types/sdk.go [hunk 1]
```diff
@@ -52,38 +27,3 @@ func DefaultConfigSetup(cfg config.Config, sdkConfig *sdk.Config) {
...
-		feegrantmodule.AppModuleBasic{},
-		upgrade.AppModuleBasic{},
-		evidence.AppModuleBasic{},
-		authzmodule.AppModuleBasic{},
-		groupmodule.AppModuleBasic{},
-		vesting.AppModuleBasic{},
-		nftmodule.AppModuleBasic{},
-		consensus.AppModuleBasic{},
-	)
-)
```
Comment (dadamu) [inline cmd/parse/types/sdk.go:89]: Remove it, the Codec should be manually defined by users who wants to use local mode node.

File: database/database.go [hunk 2]
```diff
@@ -49,7 +48,7 @@ type Database interface {
 
 	// SaveMessage stores a single message.
 	// An error is returned if the operation fails.
-	SaveMessage(msg *types.Message) error
+	SaveMessage(height int64, txHash string, msg types.Message, addresses []string) error
```
Comment (dadamu) [inline database/database.go:51]: Message is unmarshaled from TxBody which does not contain `height`, `txHash` and `addresses` information, so I change the interface to include them.

**Review by ghost — APPROVED** [review]
LGTM.

@dadamu just for me to understand better could you explain:
- What's a codec regarding transactions?
- How do you decide the data structure/layout your mapping to?

**Comment by dadamu** [comment]
@0x7u Thanks for the review!
> What's a codec regarding transactions?

Currently, Cosmos-SDK handles the data in ProtoBuffer format. In addition, Codec is the encoder/decoder for ProtoBuffer, which is responsible to decode the raw data in ProtoBuffer format into the golang structure. This [document](https://docs.cosmos.network/main/learn/advanced/encoding) tells more information.

Currently, in order to transform all the message from ProtoBuffer into JSON to store inside `messages` table, it is required to register all needed ProtoBuffer messages into Codec. But it is hard to maintain the Codec register since it is not easy to track what custom messages being used by a chain. Say, there are custom messages `MsgSaveProfile`, `MsgCreatePost` and etc., in Desmos, while there are `MsgClaimStray`, `MsgAttest` and etc. ins Jackal. So this PR tries to get JSON-parsed response from API server instead of ProtoBuffer encoded response from gRPC server so that Juno can get rid of Codec register issue.

> How do you decide the data structure/layout your mapping to?

The structure/layout is following the [Cosmos-SDK spec](https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/tx/v1beta1/tx.proto#L96). Although the original structure is based on ProtoBuffer, so some `Any` type data is required to be rewritten into JSON format. The reference is [here](https://protobuf.dev/programming-guides/proto3/#json).

**Review by MonikaCat — APPROVED** [review]
utACK

**Pull request merged**